javascript/contributeRoute.js

```diff
@@ -25,24 +24,53 @@ var geoURL = function(str, cty, ste) {
     return geoURLResponse;
 }
 
-var initMap = function(location, zoomLevel) {
+var initMap = function(location, title, content, zoomLevel) {
     map = new google.maps.Map(mapContainer, {
         center: location,
         zoom: zoomLevel,
     });
-    addPOIMarker(location, map);
+    addPOIMarker(location, title, content);
     google.maps.event.addListener(map, function() {
         addPOIMarker();
+        fitBounds();
         });
         
     return map;
 }
 
-var addPOIMarker = function(poi) {
+var addPOIMarker = function(poi, poiTitle, poiContent) {
     var marker = new google.maps.Marker({
         position: poi,
         map: map,
+        title: poiTitle.value,
     });
+    var poiContainer = document.createElement('div');
+
+    var contentDiv = document.createElement('h4');
+    contentDiv.classList.add('poi-content');
+    contentDiv.textContent = poiContent;
+
+    var titleDiv = document.createElement('h2');
+    titleDiv.classList.add('poi-title');
+    titleDiv.textContent = poiTitle;
+
+    poiContainer.appendChild(titleDiv);
+    poiContainer.appendChild(contentDiv);
+
+    var infowindow = new google.maps.InfoWindow({
+        content: poiContainer.innerHTML,
```

Comment:
could you use .textContent in place of .innerHTML ?

Reply:
not if we want the option of styling those elements unfortunately (at least I think).

Reply:
hmmm couldn't we add a CSS class to those elements either way, though?  e.g., element.classList.add("mystyle");

Reply:
The problem is that it would style the DOM, but it wouldn't style the content of the infobox because that class info would be stripped away if we just set it to the text.  I'll play in the inspector for a second to see if there is some other way to style the content.

Reply:
you're right.  I was being silly.  will update and recommit.

Reply:
I actually can get it to work that way.   I don't think there is any equivalent way to the DOM method of .textContent you can put on a string value.  the infobox takes html information as a string and uses it.  creating the DOM element and then setting it to the inner HTML is the way I get access to the method the neuters and script additions.

Reply:
Is this blocking you? We should try to reduce any innerHTML but can come back to it later.

Reply:
a little bit.  if someone wants to clone the branch and try and fix it, they can.   I'd like to push it and maybe someone can take that on as a security challenge later.

Reply:
I think that's a good solution for the time being.